### PR TITLE
tweaks

### PR DESCRIPTION
**Pull request by jerguslejko**
this back ports few changes from `remote-ml` repo that are generally applicable

### CODE REVIEW

File: {{ cookiecutter.app_name }}/.flake8 [hunk 1]
```diff
@@ -2,3 +2,4 @@
 statistics = True
 max-line-length = 120
 exclude = .git,__pycache__,.pytest_cache
+ignore = B008, W503
```
Comment (lachtanek) [inline {{ cookiecutter.app_name }}/.flake8:5]: why `B008`?

File: {{ cookiecutter.app_name }}/Dockerfile [hunk 2]
```diff
@@ -22,22 +25,22 @@ WORKDIR /app
...
 FROM base as dev
 
-RUN poetry install
+# ...
 
 ########################################################
 FROM base as production
 
-# ...
+RUN poetry install --no-dev
```
Comment (lachtanek) [inline {{ cookiecutter.app_name }}/Dockerfile:46]: why this change? this will mean all packages are installed, because I expect `install --no-dev` will not remove previously installed ones, even if they no longer belong
even if it removed them, this'd just increase the image size, as you need to push the previous layer which contained all dependencies

File: {{ cookiecutter.app_name }}/{{ cookiecutter.app_name }}/config.py [hunk 3]
```diff
@@ -9,7 +9,7 @@ class Settings(BaseSettings):
     app_name: str = "{{ cookiecutter.app_name }}"
     app_env: Literal["production", "staging", "dev", "test"] = "dev"
     release_version: Optional[str] = None
-    log_level: Literal["ERROR", "WARNING", "INFO", "DEBUG"] = "ERROR"
+    log_level: Literal["error", "warning", "info", "debug"] = "error"
```
Comment (lachtanek) [inline {{ cookiecutter.app_name }}/{{ cookiecutter.app_name }}/config.py:12]: this needs to be uppercase, lowercase won't work afaik
```
In [4]: logging.getLogger(__name__).setLevel("warning")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [4], in <cell line: 1>()
----> 1 logging.getLogger(__name__).setLevel("warning")

File /usr/lib/python3.10/logging/__init__.py:1452, in Logger.setLevel(self, level)
   1448 def setLevel(self, level):
   1449     """
   1450     Set the logging level of this logger.  level must be an int or a str.
   1451     """
-> 1452     self.level = _checkLevel(level)
   1453     self.manager._clear_cache()

File /usr/lib/python3.10/logging/__init__.py:198, in _checkLevel(level)
    196 elif str(level) == level:
    197     if level not in _nameToLevel:
--> 198         raise ValueError("Unknown level: %r" % level)
    199     rv = _nameToLevel[level]
    200 else:

ValueError: Unknown level: 'warning'

In [5]: logging.getLogger(__name__).setLevel("WARNING")

In [6]: 

```

File: {{ cookiecutter.app_name }}/{{ cookiecutter.app_name }}/logging.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 import sentry_sdk
 from parrottools.logging import configure_logging as configure
+from parrottools.logging import log_context  # noqa
```
Comment (lachtanek) [inline {{ cookiecutter.app_name }}/{{ cookiecutter.app_name }}/logging.py:5]: why? it is not used
if we are just importing it to re-export, include it in `__all__` (see https://peps.python.org/pep-0008/#public-and-internal-interfaces)

File: {{ cookiecutter.app_name }}/.flake8 (hunk 1 above)
Comment (jerguslejko) [reply {{ cookiecutter.app_name }}/.flake8:5]: fastapi does stuff like 

```
@app.get("/")
def my_handler(a = Depends(my_fn)):
    ...
```

and the `Depends` is the issue. but I will remove this from the template and only add in `remote-ml`

File: {{ cookiecutter.app_name }}/{{ cookiecutter.app_name }}/config.py (hunk 3 above)
Comment (jerguslejko) [reply {{ cookiecutter.app_name }}/{{ cookiecutter.app_name }}/config.py:12]: changed

File: {{ cookiecutter.app_name }}/.flake8 (hunk 1 above)
Comment (lachtanek) [reply {{ cookiecutter.app_name }}/.flake8:5]: oki, makes sense, but please include it in a comment, it's good to know why we are ignoring these things

**Review by lachtanek — APPROVED** [review]
